frontend/src/metabase/components/LastEditInfoLabel.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import styled from "styled-components";
+import { t } from "ttag";
+import moment from "moment";
+
+import { color } from "metabase/lib/colors";
+
+import { getUser } from "metabase/selectors/user";
+
+const Label = styled.span`
+  font-weight: bold;
+  color: ${color("text-medium")};
+`;
+
+function mapStateToProps(state) {
+  return {
+    user: getUser(state),
+  };
+}
+
+LastEditInfoLabel.propTypes = {
+  item: PropTypes.shape({
+    "last-edit-info": PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      email: PropTypes.string.isRequired,
+      first_name: PropTypes.string.isRequired,
+      last_name: PropTypes.string.isRequired,
+      timestamp: PropTypes.string.isRequired,
+    }).isRequired,
+  }),
+  user: PropTypes.shape({
+    id: PropTypes.number,
+  }).isRequired,
+};
+
+function formatEditorName(firstName, lastName) {
```

Comment:
I personally prefer keeping things as close to the place they're used at as possible. If a function is used in a single place and is living inside some shared directory like `utils`, it makes you think it's something widespread and you should be extra careful when changing it + most likely it won't be easy to move or delete. On the other hand, if the function is right there, it's clear there is only a single place it's used at and you know if you break something, it's going to happen in one single place. If we need to repeat the same thing in a different place, it can be a good reason to move a function inside `utils` I guess. Does it make sense?

Reply:
@kulyk the downside of keeping utility functions in the component files is that we will never know when other components use something identical, or very similar.

So we may end up with several variations of the same functionality in many places.

Also, this function is not unit tested, and having it declared in the component file discourages us from writing unit tests for it.

The function may seem trivial to you now, but to people just landing on the codebase and trying to see things in a larger context, it will not be trivial.

So having the unit tests will serve as descriptions of the functionality to future readers, as well as "double-entry bookkeeping".

So if we cover things nicely with unit tests we can trust that if you change something, and it breaks something far away from it, the tests will catch it.

Reply:
> the downside of keeping utility functions in the component files is that we will never know when other components use something identical or very similar.

I'm worried if we begin moving every non-UI and non-business-logic function to shared utils, we're going to end up with a huge shared library, and it won't help following DRY either. I guess it should better be solved by internal library docs and code review mostly.

> Also, this function is not unit tested, and having it declared in the component file discourages us from writing unit tests for it.

I think we should better test the end-behavior rather than a small utility like that. In that way, we would avoid rewriting tests each time the implementation is changed. I.e., we can use react-testing-library to ensure the name is displayed like "full_last_name first_name_first_char". The formatting can happen inside the component, with some function in the same file or with a function from shared utils, the test will pass in any case

Reply:
Ok, thanks.